addons/website_form/static/src/snippets/s_website_form/000.js

```diff
@@ -192,17 +193,35 @@ odoo.define('website_form.s_website_form', function (require) {
                         successMode = successPage ? 'redirect' : 'nothing';
                     }
                     switch (successMode) {
-                        case 'redirect':
-                            successPage = successPage.startsWith("/#") ? successPage.slice(1) : successPage;
+                        case 'redirect': {
+                            const hashIndex = successPage.indexOf("#");
+                            if (hashIndex > 0) {
+                                // URL containing an anchor detected: extract
+                                // the anchor from the URL if the URL is the
+                                // same as the current page URL so we can scroll
+                                // directly to the element (if found) later
+                                // instead of redirecting.
+                                let currentUrlPath = window.location.pathname;
+                                if (!currentUrlPath.endsWith("/")) {
+                                    currentUrlPath = currentUrlPath + "/";
+                                }
+                                if ([successPage, "/" + session.lang_url_code + successPage].some(link => link.startsWith(currentUrlPath + '#'))) {
+                                    successPage = successPage.substring(hashIndex);
+                                }
```

Comment:
@rdeodoo I tested the forward-port in V15 and it seems to be an issue in the code here..
I don't remember why we add the "/" after `currentUrlPath` ?
Because this is the problem... `successPage` doesn't have a "/" before the "#" when the anchor is copied from the anchor option clipboard... So, for instance, `/contactus#Text ` is never egal to `/contactus/#Text`.
Maybe both cases should be handled ? `succesPage` could be `/contactus/#Text` or `/contactus#Text` depends if the user copy/paste it from the anchor option clipboard or write it by himself with the "/" before the "#" ...
I could make the modification in the V15 forward-port and create a PR to fix it in V14...

Reply:
I think it's following https://github.com/odoo/odoo/pull/82059#discussion_r839674916
Ideally fix 14.0 yeah but start with forward port

Reply:
@rdeodoo I fixed it in the forward-port (V15.0) and created a PR to fix the V14.0

PR for V14.0 = > https://github.com/odoo/odoo/pull/120907